diff --git a/cc/scheduler/scheduler.cc b/cc/scheduler/scheduler.cc
index 9ccc5e622a..c79364aa45 100644
--- a/cc/scheduler/scheduler.cc
+++ b/cc/scheduler/scheduler.cc
@@ -13,6 +13,7 @@
 #include "base/feature_list.h"
 #include "base/functional/bind.h"
 #include "base/location.h"
+#include "base/logging.h"
 #include "base/metrics/histogram_macros.h"
 #include "base/rand_util.h"
 #include "base/task/delay_policy.h"
@@ -934,6 +935,7 @@ void Scheduler::ProcessScheduledActions() {
       case SchedulerStateMachine::Action::NONE:
         break;
       case SchedulerStateMachine::Action::SEND_BEGIN_MAIN_FRAME:
+        LOG(INFO) << "Scheduler processing SEND_BEGIN_MAIN_FRAME action";
         compositor_timing_history_->WillBeginMainFrame(begin_main_frame_args_);
         compositor_frame_reporting_controller_->WillBeginMainFrame(
             begin_main_frame_args_);
diff --git a/cc/scheduler/scheduler_state_machine.cc b/cc/scheduler/scheduler_state_machine.cc
index 8cfa825188..5fb472f411 100644
--- a/cc/scheduler/scheduler_state_machine.cc
+++ b/cc/scheduler/scheduler_state_machine.cc
@@ -9,6 +9,7 @@
 #include "base/check_op.h"
 #include "base/feature_list.h"
 #include "base/format_macros.h"
+#include "base/logging.h"
 #include "base/notreached.h"
 #include "base/time/time.h"
 #include "base/trace_event/trace_event.h"
@@ -802,8 +803,10 @@ bool SchedulerStateMachine::ShouldInvalidateLayerTreeFrameSink() const {
 }
 
 SchedulerStateMachine::Action SchedulerStateMachine::NextAction() const {
-  if (ShouldSendBeginMainFrame())
+  if (ShouldSendBeginMainFrame()) {
+    LOG(INFO) << "Impl Sends BeginMainFrame";
     return Action::SEND_BEGIN_MAIN_FRAME;
+  }
   if (ShouldRunPostCommit())
     return Action::POST_COMMIT;
   if (ShouldActivateSyncTree())
@@ -1389,6 +1392,8 @@ void SchedulerStateMachine::OnBeginImplFrame(const viz::BeginFrameArgs& args) {
   did_invalidate_layer_tree_frame_sink_ = false;
   did_perform_impl_side_invalidation_ = false;
   waiting_for_scroll_event_ = false;
+  needs_begin_main_frame_ = true;
+  LOG(INFO) << "BeginFrame Impl";
 }
 
 void SchedulerStateMachine::OnBeginImplFrameDeadline() {
diff --git a/cc/trees/layer_tree_host.cc b/cc/trees/layer_tree_host.cc
index 6bfcfba051..a426006d02 100644
--- a/cc/trees/layer_tree_host.cc
+++ b/cc/trees/layer_tree_host.cc
@@ -391,6 +391,7 @@ void LayerTreeHost::BeginMainFrameNotExpectedUntil(base::TimeTicks time) {
 
 void LayerTreeHost::BeginMainFrame(const viz::BeginFrameArgs& args) {
   DCHECK(IsMainThread());
+  LOG(INFO) << "BeginMainFrame received in LayerTreeHost";
   client_->BeginMainFrame(args);
 }
 
diff --git a/cc/trees/proxy_impl.cc b/cc/trees/proxy_impl.cc
index 01c851053f..67e81cfd72 100644
--- a/cc/trees/proxy_impl.cc
+++ b/cc/trees/proxy_impl.cc
@@ -22,6 +22,7 @@
 #include "base/trace_event/traced_value.h"
 #include "base/trace_event/typed_macros.h"
 #include "base/types/optional_ref.h"
+#include "base/logging.h"
 #include "cc/base/devtools_instrumentation.h"
 #include "cc/base/features.h"
 #include "cc/benchmarks/benchmark_instrumentation.h"
@@ -740,6 +741,7 @@ void ProxyImpl::WillNotReceiveBeginFrame() {
 
 void ProxyImpl::ScheduledActionSendBeginMainFrame(
     const viz::BeginFrameArgs& args) {
+  LOG(INFO) << "ScheduledActionSendBeginMainFrame received in ProxyImpl";
   DCHECK(IsImplThread());
 
   benchmark_instrumentation::ScopedBeginFrameTask begin_frame_task(
diff --git a/cc/trees/proxy_main.cc b/cc/trees/proxy_main.cc
index 72990d3d71..ac975884ab 100644
--- a/cc/trees/proxy_main.cc
+++ b/cc/trees/proxy_main.cc
@@ -21,6 +21,7 @@
 #include "base/trace_event/trace_id_helper.h"
 #include "base/trace_event/traced_value.h"
 #include "base/types/optional_ref.h"
+#include "base/logging.h"
 #include "cc/base/completion_event.h"
 #include "cc/base/devtools_instrumentation.h"
 #include "cc/base/features.h"
@@ -138,6 +139,7 @@ void ProxyMain::DidCompletePageScaleAnimation() {
 
 void ProxyMain::BeginMainFrame(
     std::unique_ptr<BeginMainFrameAndCommitState> begin_main_frame_state) {
+  LOG(INFO) << "BeginMainFrame received in ProxyMain";
   DCHECK(IsMainThread());
   DCHECK_EQ(NO_PIPELINE_STAGE, current_pipeline_stage_);
   // Record the final status, subsampled. Use an RAII object as this function
diff --git a/third_party/blink/renderer/core/frame/web_frame_widget_impl.cc b/third_party/blink/renderer/core/frame/web_frame_widget_impl.cc
index 580ac00119..2cc944f117 100644
--- a/third_party/blink/renderer/core/frame/web_frame_widget_impl.cc
+++ b/third_party/blink/renderer/core/frame/web_frame_widget_impl.cc
@@ -38,6 +38,7 @@
 #include "base/debug/crash_logging.h"
 #include "base/debug/dump_without_crashing.h"
 #include "base/functional/callback_helpers.h"
+#include "base/logging.h"
 #include "base/functional/function_ref.h"
 #include "base/metrics/histogram_functions.h"
 #include "base/metrics/histogram_macros.h"
@@ -2632,6 +2633,7 @@ void WebFrameWidgetImpl::OnCommitRequested() {
 }
 
 void WebFrameWidgetImpl::BeginMainFrame(const viz::BeginFrameArgs& args) {
+  LOG(INFO) << "BeginMainFrame received in Blink";
   base::TimeTicks last_frame_time = args.frame_time;
   TRACE_EVENT1("blink", "WebFrameWidgetImpl::BeginMainFrame", "frameTime",
                last_frame_time);
diff --git a/third_party/blink/renderer/platform/widget/compositing/layer_tree_view.cc b/third_party/blink/renderer/platform/widget/compositing/layer_tree_view.cc
index c32c90567f..69943776cf 100644
--- a/third_party/blink/renderer/platform/widget/compositing/layer_tree_view.cc
+++ b/third_party/blink/renderer/platform/widget/compositing/layer_tree_view.cc
@@ -273,6 +273,7 @@ void LayerTreeView::BeginMainFrame(const viz::BeginFrameArgs& args) {
   if (!delegate_)
     return;
   widget_scheduler_->WillBeginFrame(args);
+  LOG(INFO) << "BeginMainFrame received in LayerTreeView";
   delegate_->BeginMainFrame(args);
 }
 
