diff --git a/cc/scheduler/scheduler_state_machine.cc b/cc/scheduler/scheduler_state_machine.cc
index 8cfa825188..5fb472f411 100644
--- a/cc/scheduler/scheduler_state_machine.cc
+++ b/cc/scheduler/scheduler_state_machine.cc
@@ -9,6 +9,7 @@
 #include "base/check_op.h"
 #include "base/feature_list.h"
 #include "base/format_macros.h"
+#include "base/logging.h"
 #include "base/notreached.h"
 #include "base/time/time.h"
 #include "base/trace_event/trace_event.h"
@@ -802,8 +803,10 @@ bool SchedulerStateMachine::ShouldInvalidateLayerTreeFrameSink() const {
 }
 
 SchedulerStateMachine::Action SchedulerStateMachine::NextAction() const {
-  if (ShouldSendBeginMainFrame())
+  if (ShouldSendBeginMainFrame()) {
+    LOG(INFO) << "Impl Sends BeginMainFrame";
     return Action::SEND_BEGIN_MAIN_FRAME;
+  }
   if (ShouldRunPostCommit())
     return Action::POST_COMMIT;
   if (ShouldActivateSyncTree())
@@ -1389,6 +1392,8 @@ void SchedulerStateMachine::OnBeginImplFrame(const viz::BeginFrameArgs& args) {
   did_invalidate_layer_tree_frame_sink_ = false;
   did_perform_impl_side_invalidation_ = false;
   waiting_for_scroll_event_ = false;
+  needs_begin_main_frame_ = true;
+  LOG(INFO) << "BeginFrame Impl";
 }
 
 void SchedulerStateMachine::OnBeginImplFrameDeadline() {
