diff --git a/third_party/blink/renderer/core/dom/node.cc b/third_party/blink/renderer/core/dom/node.cc
index c580313c33..5426191d2c 100644
--- a/third_party/blink/renderer/core/dom/node.cc
+++ b/third_party/blink/renderer/core/dom/node.cc
@@ -2685,12 +2685,12 @@ String Node::ToString() const {
   return builder.ReleaseString();
 }
 
-#if DCHECK_IS_ON()
-
 String Node::ToTreeStringForThis() const {
   return ToMarkedTreeString(this, "*");
 }
 
+#if DCHECK_IS_ON()
+
 String Node::ToFlatTreeStringForThis() const {
   return ToMarkedFlatTreeString(this, "*");
 }
@@ -2746,6 +2746,8 @@ void Node::PrintNodePathTo(std::ostream& stream) const {
   }
 }
 
+#endif
+
 static void AppendMarkedTree(const String& base_indent,
                              const Node* root_node,
                              const Node* marked_node1,
@@ -2829,6 +2831,8 @@ static void AppendMarkedTree(const String& base_indent,
   }
 }
 
+#if DCHECK_IS_ON()
+
 static void AppendMarkedFlatTree(const String& base_indent,
                                  const Node* root_node,
                                  const Node* marked_node1,
@@ -2855,6 +2859,8 @@ static void AppendMarkedFlatTree(const String& base_indent,
   }
 }
 
+#endif
+
 String Node::ToMarkedTreeString(const Node* marked_node1,
                                 const char* marked_label1,
                                 const Node* marked_node2,
@@ -2872,6 +2878,8 @@ String Node::ToMarkedTreeString(const Node* marked_node1,
   return builder.ReleaseString();
 }
 
+#if DCHECK_IS_ON()
+
 String Node::ToMarkedFlatTreeString(const Node* marked_node1,
                                     const char* marked_label1,
                                     const Node* marked_node2,
@@ -3768,6 +3776,13 @@ void Node::Trace(Visitor* visitor) const {
 
 }  // namespace blink
 
+void ShowTree(const blink::Node* node) {
+  if (node)
+    LOG(INFO) << "\n" << node->ToTreeStringForThis().Utf8();
+  else
+    LOG(INFO) << "Cannot showTree for <null>";
+}
+
 #if DCHECK_IS_ON()
 
 void ShowNode(const blink::Node* node) {
@@ -3777,13 +3792,6 @@ void ShowNode(const blink::Node* node) {
     LOG(INFO) << "Cannot showNode for <null>";
 }
 
-void ShowTree(const blink::Node* node) {
-  if (node)
-    LOG(INFO) << "\n" << node->ToTreeStringForThis().Utf8();
-  else
-    LOG(INFO) << "Cannot showTree for <null>";
-}
-
 void ShowNodePath(const blink::Node* node) {
   if (node) {
     std::stringstream stream;
diff --git a/third_party/blink/renderer/core/dom/node.h b/third_party/blink/renderer/core/dom/node.h
index c637e01b20..012dc58331 100644
--- a/third_party/blink/renderer/core/dom/node.h
+++ b/third_party/blink/renderer/core/dom/node.h
@@ -900,14 +900,14 @@ class CORE_EXPORT Node : public EventTarget {
 
   String ToString() const;
 
-#if DCHECK_IS_ON()
   String ToTreeStringForThis() const;
-  String ToFlatTreeStringForThis() const;
-  void PrintNodePathTo(std::ostream&) const;
   String ToMarkedTreeString(const Node* marked_node1,
                             const char* marked_label1,
                             const Node* marked_node2 = nullptr,
                             const char* marked_label2 = nullptr) const;
+#if DCHECK_IS_ON()
+  String ToFlatTreeStringForThis() const;
+  void PrintNodePathTo(std::ostream&) const;
   String ToMarkedFlatTreeString(const Node* marked_node1,
                                 const char* marked_label1,
                                 const Node* marked_node2 = nullptr,
@@ -1398,10 +1398,11 @@ CORE_EXPORT std::ostream& operator<<(std::ostream&, const Node*);
 
 }  // namespace blink
 
+void ShowTree(const blink::Node*);
+
 #if DCHECK_IS_ON()
 // Outside the blink namespace for ease of invocation from gdb.
 void ShowNode(const blink::Node*);
-void ShowTree(const blink::Node*);
 void ShowNodePath(const blink::Node*);
 #endif
 
