diff --git a/third_party/blink/renderer/core/paint/pre_paint_tree_walk.cc b/third_party/blink/renderer/core/paint/pre_paint_tree_walk.cc
index c2589ef87b..8c137b87fe 100644
--- a/third_party/blink/renderer/core/paint/pre_paint_tree_walk.cc
+++ b/third_party/blink/renderer/core/paint/pre_paint_tree_walk.cc
@@ -4,9 +4,14 @@
 
 #include "third_party/blink/renderer/core/paint/pre_paint_tree_walk.h"
 
+#include "base/command_line.h"
 #include "base/debug/dump_without_crashing.h"
+#include "base/logging.h"
+#include "base/time/time.h"
 #include "base/types/optional_util.h"
+#include "third_party/blink/renderer/core/dom/document.h"
 #include "third_party/blink/renderer/core/dom/document_lifecycle.h"
+#include "third_party/blink/renderer/core/dom/node.h"
 #include "third_party/blink/renderer/core/editing/frame_selection.h"
 #include "third_party/blink/renderer/core/frame/event_handler_registry.h"
 #include "third_party/blink/renderer/core/frame/local_dom_window.h"
@@ -69,8 +74,23 @@ void PrePaintTreeWalk::WalkTree(LocalFrameView& root_frame_view) {
     return;
   }
 
-  DCHECK_EQ(root_frame_view.GetFrame().GetDocument()->Lifecycle().GetState(),
-            DocumentLifecycle::kInPrePaint);
+  Document* document = root_frame_view.GetFrame().GetDocument();
+  if (!document) {
+    return;
+  }
+
+  DCHECK_EQ(document->Lifecycle().GetState(), DocumentLifecycle::kInPrePaint);
+
+  if (base::CommandLine::ForCurrentProcess()->GetSwitchValueASCII("type") ==
+          "renderer" &&
+      document->IsInMainFrame()) {
+    const int64_t timestamp_ms =
+        base::TimeTicks::Now().since_origin().InMilliseconds();
+    LOG(INFO) << "PrePaint URL: " << document->Url().GetString().Utf8()
+              << " State: " << document->Lifecycle().GetState()
+              << " timestamp (ms): " << timestamp_ms;
+    ShowTree(document);
+  }
 
   PrePaintTreeWalkContext context;
 
